impeller/renderer/backend/gles/reactor_gles.cc

```diff
@@ -23,7 +72,13 @@ ReactorGLES::ReactorGLES(std::unique_ptr<ProcTableGLES> gl)
   is_valid_ = true;
 }
 
-ReactorGLES::~ReactorGLES() = default;
+ReactorGLES::~ReactorGLES() {
+  for (auto& handle : handles_) {
```

Comment:
This needs to be guarded by a `CanReactOnCurrentThread`. Otherwise, there is no guarantee that the thread can actually perform OpenGL operations. Perhaps there should be a shutdown method that can be explicitly called in addition to being called from the destructor. If there are pending handles and there can be no reactions on the current thread, then we could just log.

Reply:
Added a `CanReactOnCurrentThread` check.

Unfortunately, there does not seem to be an obvious place for the shell to clean up the `ReactorGLES` on a reactor worker thread during shutdown.  The `ReactorGLES` is not deleted until the `PlatformView` is deleted on the platform thread in the final phase of shell shutdown.

The shell will be relying on destruction of the EGL context to clean up these objects.